PokemonTabletopAdventures_v3/PTA3.html

```diff
@@ -2757,4 +2783,229 @@ <h2>Configuration Options</h2>
+      if(json){
+        const newCharacterData = parseCharacterData(json);
+        setAttrs(newCharacterData);
+      } else {
+        setAttrs({ ["json_import_response"]: "Error - JSON string appears invalid" });
+      }
+    });
+  });
+
+  on("clicked:generate-export", function() {
+    getSectionIDs("repeating_moves", function (moveIDs) {
+      getSectionIDs("repeating_features", function (featureIDs) {
+        const idmoves = moveIDs.flatMap(id =>
+            r20export_moves.map(moveAttr => `repeating_moves_${id}_${moveAttr}`)
+        );
+        const idfeatures = featureIDs.flatMap(id =>
+            r20export_features.map(featureAttr => `repeating_features_${id}_${featureAttr}`)
+        );
+        getAttrs([...r20export_attrs, ...idmoves, ...idfeatures], function (values) {
+          let topLevelAttrs = includeOnlyKeysInArray(values, r20export_attrs);
+          let moveAttrs = includeOnlyKeysInArray(values, idmoves);
+          let featureAttrs = includeOnlyKeysInArray(values, idfeatures);
+          let chunkedMoves = partitionObject(moveAttrs, r20export_moves);
+          let chunkedFeatures = partitionObject(featureAttrs, r20export_features);
+          topLevelAttrs["exportmoves"] = chunkedMoves;
+          topLevelAttrs["exportfeatures"] = chunkedFeatures;
+          topLevelAttrs["Source"] = "roll20 [" + new Date().toUTCString() + "]";
+          setAttrs({
+            "json_export": JSON.stringify(topLevelAttrs, null, 2)
+          });
+        });
+      });
+    });
+  });
+
+  function includeOnlyKeysInArray(sourceObject, filterArray) {
+    return _.pick(sourceObject, ...filterArray);
+  }
+
+  //This relies on elements in the object being in the correct order, which seems to be true always?
+  function partitionObject(sourceObject, baseKeys) {
+    const sourceArray = Object.entries(sourceObject);
+    const chunkSize = baseKeys.length;
+    const chunkCount = sourceArray.length / chunkSize;
+    let results = [];
+    let totalCount = 0;
+    for (chunk = 0; chunk < chunkCount; chunk++) {
+      let partition = {};
+      for(count = 0; count < chunkSize; count++){
+        const baseName = baseKeys[count];
+        partition[baseName] = sourceArray[totalCount][1];
+        totalCount++;
+      }
+      results.push(partition);
+    }
+    return results;
+  }
+
+  function tryParseJson(jsonString) {
+    // https://stackoverflow.com/a/20392392
+    try {
+      var o = JSON.parse(jsonString);
+      if (o && typeof o === "object") {
+          return o;
+      }
+    } catch (e) { }
+    return false;
+  }
+
+  function parseCharacterData(characterData) {
+    const source = characterData["Source"];
+    if(!source) return { ["json_import_response"]: "Error - Unable to identify source of JSON data" };
+    if(source.startsWith("Pokelicious")) {
+      // Identify as a Pokelicious sheet
+      let response = { ["json_import_response"]: "Error - Unable to parse all attributes" }
```

Comment:
Should be able to replace this with a `const`. In its infinite wisdom, JS lets us modify constant objects, just not reassign them.
```suggestion
      const response = { ["json_import_response"]: "Error - Unable to parse all attributes" }
```